services/schemas/duty.zschema.ts

```diff
@@ -0,0 +1,29 @@
+import { z } from 'zod';
+import { ObjectId } from 'mongodb';
+
+const IdSchema = z.preprocess(
+  (id: any) => new ObjectId(id),
+  z.custom<ObjectId>().or(z.string()),
+);
+
+const DutySchema = z.object({
+  _id: IdSchema.optional(),
```

Comment:
No need to have an id in the schema, let mongoDb generate one for you

Reply:
resolved